packages/core/src/utils/operation.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  GraphQLRequest,
+  Operation,
+  OperationContext,
+  OperationType,
+} from '../types';
+import { Warning, deprecationWarning } from './deprecation';
+
+// TODO: Remove when the deprecated `operationName` property is removed
+const DEPRECATED: Record<string, Warning> = {
+  operationName: {
+    key: 'Operation.operationName',
+    message:
+      'The "Operation.operationName" property has been deprecated and will be removed in a future release of urql. Use "Operation.kind" instead.',
+  },
+};
+
+export const makeOperation = (
+  kind: OperationType,
+  {
+    key,
+    query,
+    variables,
+  }: Pick<GraphQLRequest, 'key' | 'query' | 'variables'>,
+  context: OperationContext
+): Operation => ({
+  key,
+  query,
+  variables,
+  kind,
+  context,
```

Comment:
```suggestion
  kind: OperationType,
  request: GraphQLRequest,
  context: OperationContext
): Operation => ({
  kind,
  key: request.key,
  query: request.query,
  variables: request.variables,
  context,
```

Rearranging here; also putting `kind` first for debugging. I really like this pattern :+1: pretty unintrusive

Reply:
Thanks! I used the `Pick<>` with `GraphQLRequest` so that I could also pass an existing Operation there that I want to copy. For example, if you have an existing query Operation that you want to issue a 'teardown' for:

```ts
makeOperation('teardown', query, query.context)
```

Reply:
I can definitely see how it's more readable without the destructuring, though. I'll switch that back. :+1:

Reply:
I think the `Pick` isn't necessary though there, right? An `Operation` already extends `GraphQLRequest` so it should also be accepted there without `Pick` as interfaces in TS are open types

Reply:
Oh, good point!